app/src/main/java/org/xbmc/kore/utils/Task.java

```diff
@@ -0,0 +1,81 @@
+package org.xbmc.kore.utils;
+
+import android.support.annotation.NonNull;
+
+/**
+ * Represents a computation that will eventually complete.
+ *
+ * @param <T> The type of the value being computed or awaited.
+ */
+public interface Task<T> {
+
+    /**
+     * Starts the computation, invokes a procedure when done.
+     *
+     * @param then The consumer of the value being computed.
+     */
+    void start(@NonNull OnFinish<? super T> then);
+
+    /**
+     * A procedure that awaits a value.
+     *
+     * @param <T> The type of the value generated by a previous computation.
```

Comment:
Shouldn't this be the same type of the Task that implements the `OnFinish`? If so, the comment can be removed or should make that clear.